staging/src/k8s.io/legacy-cloud-providers/azure/azure_routes.go

```diff
@@ -184,6 +184,22 @@ func (az *Cloud) CreateRoute(ctx context.Context, clusterName string, nameHint s
 		},
 	}
 
+	actualRoute, err := az.RoutesClient.Get(ctx, az.RouteTableResourceGroup, az.RouteTableName, routeName)
```

Comment:
could you switch to `az.ListRoutes()` here? `ListRoutes()` would get the routes from cache, which would help to reduce the chances of throttling issues.

Reply:
sure, make more sense.